### PR TITLE
Revise backend to enable more control over preamble

### PR DESCRIPTION
**Pull request by TotalVerb**
I added some more advanced package dependency analysis, so that graphicx and texments are only used when needed. I added the `document` function to allow more control over the document. In particular, articles are now supported, and LaTeX.jl will automatically use the correct section tags.

These changes have been documented and there is some test coverage added.

### CODE REVIEW

File: src/LaTeX.jl [hunk 1]
```diff
@@ -3,11 +3,20 @@
 module LaTeX
 
 using SHA
+using Iterators
```
Comment (rened) [inline src/LaTeX.jl]: Is this really used in the code? I couldn't find any use of it?

**Comment by rened** [comment]
Thanks a lot for this great contribution! I wanted to add this functionality for a long time...

Looks good to merge to me, just wondering whether `Iterators` is really used / needed.

**Comment by TotalVerb** [comment]
Thank you for the feedback! `Iterators` was used, but only once, and it is definitely not needed, so I've removed that part of this pull request.

**Comment by rened** [comment]
@TotalVerb thanks again! great stuff!

**Pull request merged**